string.Format: fixed How arguments are formatted section

pkulikov opened the pull request:
Fixes dotnet/docs#5151

rpetrusha commented on xml/System/String.xml:
```diff
@@ -4793,15 +4793,17 @@ In .NET Core, sorting and comparison operations are based on [Version 8.0.0 of t
 -   If the argument implements the <xref:System.IFormattable> interface, its <xref:System.IFormattable.ToString%2A?displayProperty=nameWithType> implementation is called.  
   
--   The argument's parameterless `ToString` method, which is either overridden or inherited from the <xref:System.Object> class, is called.  
+-   The argument's parameterless `ToString` method, which is either overridden or inherited from a base class, is called.  
```
again a nit unrelated to your PR: is overridden is wrong; it should be "which either overrides or inherits from a base class implementation".

pkulikov commented:
@rpetrusha thank you for review; I've addressed your feedback. You've mentioned *one comment about your PR*, though I see only two corrections unrelated to PR.

rpetrusha commented on xml/System/String.xml:
```diff
@@ -4793,15 +4793,17 @@ In .NET Core, sorting and comparison operations are based on [Version 8.0.0 of t
+-   If the argument is `null`, the method inserts <xref:System.String.Empty?displayProperty=nameWithType> into the result string. You don't have to be concerned with handling a <xref:System.NullReferenceException> for null arguments. 
   
--   If you call the <xref:System.String.Format%28System.IFormatProvider%2CSystem.String%2CSystem.Object%5B%5D%29> overload and the `provider` parameter implements the <xref:System.ICustomFormatter> interface, the argument is passed to the `provider` object's <xref:System.ICustomFormatter.Format%28System.String%2CSystem.Object%2CSystem.IFormatProvider%29?displayProperty=nameWithType> method. If th ...
+-   If you call the <xref:System.String.Format%28System.IFormatProvider%2CSystem.String%2CSystem.Object%5B%5D%29> overload and the `provider` object's <xref:System.IFormatProvider.GetFormat%2A?displayProperty=nameWithType> implementation returns non-null <xref:System.ICustomFormatter> implementation, the argument is passed to its <xref:System.ICustomFormatter.Format%28System.String%2CSystem.Objec ...
```
There are two occurrences of "non-null" here; both should be preceded by a ("a non-null")

pkulikov commented:
@rpetrusha thanks! Fixed!

rpetrusha commented:
Closing and reopening to force a new license/cla check.

BillWagner commented:
merging based on @rpetrusha approval.
